dali/kernels/imgproc/color_manipulation/brightness_contrast_gpu.h

```diff
@@ -39,67 +40,40 @@ struct SampleDescriptor {
 
 
 /**
- * Coverts Roi to flattened TensorListShape
+ * Flattens the TensorShape
  *
- * Flattened TensorListShape, is a TensorListShape, which doesn't have `channels`
- * as separate dimension. Instead, Width dimension is larger by `nchannels` number of times.
- * This is because, the analysis is channel-agnostic.
+ * Flattened TensorShape in case of BrightnessContrast kernel is a TensorShape,
+ * in which channel-dimension is removed. Instead, the one-before dimension is
+ * multiplied by channel-dimension size.
  *
- * Function assumes, that memory layout is *HWC, and the Roi
- * is represented as [[x_lo, y_lo], [x_hi, y_hi]].
- * Therefore, while copying, order of values needs to be reversed.
+ * E.g. [640, 480, 3] -> [640, 1440]
+ *
+ * The reason is that BrightnessContrast calculations are channel-agnostic
+ * (the same operation is applied for every channel), therefore BlockSetup
+ * and SampleDescriptor don't need to know about channels.
  */
 template <size_t ndims>
-TensorListShape<ndims> RoiToShape(const std::vector<Roi<ndims>> &rois, int nchannels) {
-  TensorListShape<ndims> ret(rois.size());
-
-  size_t i = 0;
-  for (const auto &roi : rois) {
-    assert(all_coords(roi.hi >= roi.lo) && "Cannot create a tensor shape from an invalid Box");
-    TensorShape<ndims> ts;
-    auto e = roi.extent();
-    auto ridx = ndims;
-    ts[--ridx] = e[0] * nchannels;
-    for (size_t idx = 1; idx < ndims; idx++) {
-      ts[--ridx] = e[idx];
-    }
-    ret.set_tensor_shape(i++, ts);
+TensorShape<ndims - 1> Flatten(const TensorShape<ndims> &shape) {
+  static_assert(ndims >= 2, "If there are less than 2 dims, there's nothing to flatten...");
```

Comment:
Doesn't look formal enough
```suggestion
  static_assert(ndims >= 2, "If there are less than 2 dims, there's nothing to flatten.");
```